### DISCUSSION TITLE
WebUI scan/learn not working

### DISCUSSION THREAD

**Discussion by flaggz** [rspamd/rspamd#4188] [Q&A] [answered]
As stated in subject I can't get to work the Scan/learn in webui
when I input the mail text header I receive this on the gui

<img width="300" alt="Schermata 2022-06-06 alle 10 19 00" src="https://user-images.githubusercontent.com/7655589/172123425-3bec9fad-b542-41e9-8717-6f67493611aa.png">

and this from the log

`2022-06-06 10:12:13 #113655(controller) rspamd_controller_check_password: using password as enable_password for a privileged command
2022-06-06 10:13:28 #113655(controller) <875c08>; csession; rspamd_controller_check_fin_task: cannot check <(NULL)>: invalid command`

**Comment by moisseev** [comment]
It is not clear how you have managed that.
The log message states you are using unprivileged password, but in this case the learn part of WebUI should be hidden.
Would you post the`controller` section of the `rspamadm configdump worker` command output (with obfuscated passwords)?

**Reply by flaggz** [reply]
controller {
    password = "...";
    static_dir = "/usr/share/rspamd/www";
    bind_socket = "localhost:11334";
    secure_ip = "127.0.0.1";
    secure_ip = "::1";
    count = 1;
}

**Reply by moisseev** [reply]
I guess, you are accessing WebUI via reverse proxy?
You should configure `enable_password` and use it for privileged commands.

**Reply by flaggz** [reply]
done but same problem. used same password for password = and enable_password =

`2022-06-06 22:38:43 #678298(controller) <59834c>; csession; rspamd_controller_check_password: absent password has been specified; source ip: 127.0.0.1:57492
2022-06-06 22:38:50 #678298(controller) <45f89b>; csession; rspamd_controller_check_fin_task: cannot check <(NULL)>: invalid command`

docs states: `https://rspamd.com/doc/workers/controller.html`
` If only one password is specified in the configuration, it is used for both type of commands`

**Reply by moisseev** [reply]
> done but same problem. used same password for password = and enable_password =

The passwords have to be distinct, so the backend be able to distinguish which one was sent.

> docs states: `https://rspamd.com/doc/workers/controller.html`
> ` If only one password is specified in the configuration, it is used for both type of commands`

It doesn't work as described in the docs. Probably it is a bug.

**Reply by flaggz** [reply]
changed the enable_password differently to password but the problem remains
I'm always restarting rspamd after each change

`2022-06-07 08:55:15 #769148(controller) <a4eq6k>; controller; rspamd_check_encrypted_password: incorrect or absent password has been specified
2022-06-07 08:55:15 #769148(controller) rspamd_log_reset_repeated: Last message repeated 2 times
2022-06-07 08:55:15 #769148(controller) <13c7dc>; csession; rspamd_controller_check_fin_task: cannot check <(NULL)>: invalid command`

`controller {
    password = "$2$aq...";
    static_dir = "/usr/share/rspamd/www";
    enable_password = "$2$7m...";
    bind_socket = "localhost:11334";
    secure_ip = "127.0.0.1";
    secure_ip = "::1";
    count = 1;
}`

**Reply by moisseev** [reply]
> 2022-06-07 08:55:15 #769148(controller) <a4eq6k>; controller; rspamd_check_encrypted_password: incorrect or absent password has been specifie

Wrong password?

**Reply by flaggz** [reply]
password is good. it is connecting to the webui.

relevant apache logs
`[07/Jun/2022:10:15:21 +0200] "GET /rspamd/stat HTTP/2.0" 403 243 "https://www.mydomain/rspamd/"
[07/Jun/2022:10:15:42 +0200] "POST /rspamd/checkv2 HTTP/2.0" 400 246 "https://www.mydomain/rspamd/"
`

relevant apache config
`RewriteRule ^/rspamd$ /rspamd/ [R]
ProxyPreserveHost On
ProxyPass /rspamd http://localhost:11334/
ProxyPassReverse /rspamd http://localhost:11334/`

**Comment by flaggz** [comment, marked as answer]
problem solved
I've changed apache directive to

`RewriteRule ^/rspamd/(.*) http://127.0.0.1:11334/$1 [P,L]
<Location /rspamd >
    Options FollowSymLinks
    Require all granted
</Location>`

and now everything works
I think it should be added to the QuickStart guide too since it is documented only for nginx
